client/app/worlds/worlds.controller.js

```diff
@@ -16,7 +18,10 @@ angular.module('farseerApp')
     }
 
     $scope.convertToLowerCaseNoSpaces = function(nameToConvert) {
-      return nameToConvert.replace(/ /g, '').toLowerCase();
+      if (nameToConvert !== null && nameToConvert !== undefined) {
```

Comment:
Looking at what this method is trying to do, a 'truthy' check should be sufficient. 
Something like
if(nameToConvert){
  //code
}
will do the trick.